calyx-opt/src/analysis/live_range_analysis.rs

```diff
@@ -1373,6 +1391,21 @@ impl LiveRangeAnalysis {
 
                 (alive, gens, input_kills)
             }
+            ir::Control::Repeat(ir::Repeat {
+                body, num_repeats, ..
+            }) => {
+                let (a, g, k) =
+                    self.build_live_ranges(body, alive, gens, kills);
+                if *num_repeats == 1 {
+                    // only one execution, so we don't need to iterate through
+                    // twice
+                    (a, g, k)
+                } else {
+                    // need to feed the live nodes on the output of the body
+                    // back into the body to get correct live range analysis
+                    self.build_live_ranges(body, a, g, k)
+                }
```

Comment:
I'm a little iffy about the special handling of the `repeat 0` and `repeat 1` cases in the analyses. To me, `collapse-control` should be the thing that gets rid of trivial repeat statements and the logic in the analysis should be generic. This does have the problem that users can get potentially unpredictable results if they forget to call that pass before running the analysis. 

Maybe @sampsyo can chime on this as well?

Reply:
Yeah, that's a good point… it's not a huge sin, if handling the n=1 case in concert with the general case is just too annoying to implement! But as a somewhat weak first-order principle, it keeps things a little simpler if we try to adhere to "division of labor" when it's not in conflict with other goals.

Reply:
Ok, what I've done for now is added dynamic `repeat 0/1` as a case for `collapse-control`. 
I also changed the default pipeline to run `collapse-control` twice: once at the beginning of the `pre-opt` and once at the end. 
Live Range Analysis now doesn't make a special case for n=1 (it relies on `collapse-control` to handle these speical cases). 
For `DominationAnalysis`, I still handle the n=0 case. The reason why I want to keep it is that the domination analysis would actually be incorrect if we didn't especially handle the n=0 case... things in a repeat n=0 statement shouldn't dominate anything, and if we say they do, then that's wrong. 

Either way, lmk if you think I should change any of these. I get that what I'm doing rn may not nec be the cleanest solution.